components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2044,6 +2050,42 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return ports;
     }
 
+    public async updateRepoStatus(
+        ctx: TraceContext,
+        workspaceId: string,
+        status: Required<WorkspaceInstanceRepoStatus> | undefined,
+    ): Promise<void> {
+        traceAPIParams(ctx, { workspaceId, status });
+        traceWI(ctx, { workspaceId });
+
+        const user = await this.checkAndBlockUser("updateRepoStatus");
+
+        const workspace = await this.internalGetWorkspace(user, workspaceId, this.workspaceDb.trace(ctx));
+        let instance = await this.workspaceDb.trace(ctx).findCurrentInstance(workspaceId);
+        if (!instance) {
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, `workspace ${workspaceId} has no instance`);
+        }
+        traceWI(ctx, { instanceId: instance.id });
+        await this.guardAccess({ kind: "workspaceInstance", subject: instance, workspace }, "update");
+
+        if (WorkspaceInstanceRepoStatus.equals(instance.status.repo, status)) {
+            return;
+        }
+
+        instance = await this.workspaceDB.trace(ctx).updateInstancePartial(instance.id, {
+            status: {
+                ...instance.status,
+                repo: status,
+            },
+        });
+        await this.messagebus.notifyOnInstanceUpdate(workspace.ownerId, instance);
```

Comment:
it works, but dashboard for some reason sees one update behind. @AlexTugarev @svenefftinge maybe you know why